1xx-enable-ssl-for-kafka-bridge.md

```diff
@@ -0,0 +1,137 @@
+# Add support for TLS/SSL on the HTTP interface 
+
+This proposal adds an option to enable TLS encryption on HTTP interface so that requests from clients to the Kafka Bridge can be sent with encryption.
+
+## Current situation
+
+Strimzi does not support TLS/SSL for HTTP Bridge, so client connections to HTTP Bridge are not encrypted at all. There is a single listener for all the endpoints that are used for Kafka operations as well as admin operations such as healthcheck and monitoring and it runs on port 8080 by default.
+
+## Motivation
+
+The connection between HTTP Bridge and Kafka cluster can be secured however, currently there is no support for securing the connections to HTTP Bridge from clients. The current recommendation is to use firewalls or API gateways to secure the client connections. This requires users to make additional configurations and does not provide the best user experience. 
+
+Moreover, some of the endpoints are for internal use only, such as `/healthy`, `/ready` and `/metrics` but they are exposed on the same listener and the same port as the other endpoints that are used by external clients. This means the endpoints for internal use only are also exposed to external clients. And if SSL is enabled, it will introduce more operational complexity as these internal endpoints would need to be configured with TLS and certificates.
+
+## Proposal
+This proposal also adds new configurations to enable SSL server and specify locations of certificate and key files in PEM format so that client connections to HTTP Bridge can be encrypted. The new configurations will be added to the current `http` prefixed configurations:
+- http.ssl.enable
+- http.ssl.keystore.location
+- http.ssl.keystore.key.location
+- http.ssl.enabled.protocols
+- http.ssl.enabled.cipher.suites
+- http.admin.port
```

Comment:
It does not really have any admin options. Should we try to find some better name for it?

Reply:
Probably `http.internal.port` or `http.healthmetrics.port` makes most sense. @ppatierno WDYT?

Reply:
I see `http.healthmetrics.port` too much specific. I liked the `management` one but I guess it's going to be refused because too similar to `admin`. I would go with `internal` at this point ... even if it's not so clear what actually it provides (it just says, it's internal).

Reply:
I get the ugliness of all the alternatives we have here.  But do we have really anything better? I mean it does not do any management, or? Would it do some management one day?

Reply:
I was taking a look around for other project and it looks like Spring and Quarkus are using the `management` term for this kind of thing. AFAIU Quarkus has a `management` port exposing health, metrics, info and openapi (very similar to the ones we have) https://quarkus.io/guides/management-interface-reference and https://www.baeldung.com/quarkus-management-interface
Spring seems to expose much more still through a `management` port.
At this point, could this name be considered quite common if used by this very well know development framework? Also should we revisit the presence of /info and /openapi on it?

Reply:
Ok, I guess I can live with it if others use the same bad name as well :-). Maybe one day we will add some management to it, some dynamic reconfiguriation or something 🤷